lmdeploy/pytorch/engine/model_agent.py

```diff
@@ -655,7 +655,7 @@ def _build_model(self,
             model_init_kwargs = dict(torch_dtype=torch_dtype,
                                      device_map=device)
             model_init_kwargs.update(self.model_config.init_kwargs)
-            if self.model_config.auto_model_cls is AutoModelForCausalLM:
+            if hasattr(self.model_config.auto_model_cls, 'from_config'):
```

Comment:
I would rather add `trust_remote_code` into `model_config.init_kwargs` so we do not need to check anything here.